services/QuillLMS/engines/evidence/lib/evidence/evidence/synthetic/initial_data.rb

```diff
@@ -0,0 +1,138 @@
+# frozen_string_literal: true
+
+module Evidence
+  module Synthetic
+    class InitialData
+      Result = Struct.new(:text, :seed, keyword_init: true)
+
+      WORD_SPLIT_COUNT = 70
+      SPACE = ' '
+      BLANK = ''
+      PERIOD = '.'
+
+      FULL_COUNT = 100
+      FULL_NOUN_COUNT = 50
+      SECTION_COUNT = 50
+
+      TEMP_PASSAGE = 1
+      TEMP_SECTION = 0.5 # give a lower temp (creativity) when it has less info
+
+      attr_reader :passage, :stem, :nouns, :results
+
+      # returns a hash of the form {'csv name' => CSVString, 'csv name2' =>...}
+      def self.generate_csvs(activity_id:, nouns: [])
+        activity = Activity.find(activity_id)
+        passage = activity.passages.first.text
+        prompts = activity.prompts
+        short_name = activity.title.first(20).gsub(' ', '_')
+
+        csvs = {}
+
+        prompts.each.with_index do |prompt, index|
+          csv_name = "#{short_name}_#{prompt.conjunction}"
+
+          data = new(passage: passage, stem: prompt.text, nouns: nouns)
+          data.run
+
+          csvs[csv_name] = data.generate_csv
+          if index == 0
```

Comment:
It's not clear to me why we're using index 0 for this code branch.

Reply:
Should we just move this block to before the `each` loop using `prompts.first.text` when initializing the `data` object?

Reply:
I'll make a comment in the code. This isn't the most elegant code, but I want to generate the passage guide, it's a csv with of the text_chunk indexes and text (for help in analysis). I only need to generate this once, so I'm doing it the first time.

Reply:
@anathomical Yeah, I did something similar to make it less confusing. Thanks for the feedback.